### PR TITLE
[Intl] Cleanup

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #...   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

cleanup of #31365 for 3.4 + some other stuff to keep in sync across branches

### CODE REVIEW

File: src/Symfony/Component/Intl/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,91 +0,0 @@
-Contributing to the Intl component
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/CONTRIBUTING.md]: this is very outdated AFAIK (cc @jakzal)

Comment (jakzal) [reply src/Symfony/Component/Intl/CONTRIBUTING.md]: Indeed :)

File: src/Symfony/Component/Intl/Data/Generator/CurrencyDataGenerator.php [hunk 2]
```diff
@@ -25,37 +25,20 @@
...
+        'XBD' => true, // European Unit of Account (XBD)
+        'XUA' => true, // ADB Unit of Account
+        'XAU' => true, // Gold
+        'XAG' => true, // Silver
+        'XPT' => true, // Platinum
+        'XPD' => true, // Palladium
+        'XSU' => true, // Sucre
+        'XDR' => true, // Special Drawing Rights
+        'XTS' => true, // Testing Currency Code
+        'XXX' => true, // Unknown Currency
```
Comment (jakzal) [inline src/Symfony/Component/Intl/Data/Generator/CurrencyDataGenerator.php]: Not sure what's the point of this. Constants were self explanatory, now we need a comment.

Comment (ro0NL) [reply src/Symfony/Component/Intl/Data/Generator/CurrencyDataGenerator.php]: maintaining the consts is an extra task, not worth it IMHO. i didnt follow it for languages also.

feel free to ignore :) but it reduces noise and makes adding entries more straightforward (the consts are only used here; it's not real API needed)

File: src/Symfony/Component/Intl/Data/Provider/CurrencyDataProvider.php [hunk 3]
```diff
@@ -13,7 +13,6 @@
 
 use Symfony\Component\Intl\Data\Bundle\Reader\BundleEntryReaderInterface;
 use Symfony\Component\Intl\Exception\MissingResourceException;
-use Symfony\Component\Intl\Locale;
```
Comment (jakzal) [inline src/Symfony/Component/Intl/Data/Provider/CurrencyDataProvider.php]: Erm... are you sure about this?

Comment (ro0NL) [reply src/Symfony/Component/Intl/Data/Provider/CurrencyDataProvider.php]: yep :) 

https://github.com/symfony/symfony/blob/70a941ec0c22c45c3bf2c47dec6bbd95882be283/src/Symfony/Component/Intl/ResourceBundle/CurrencyBundleInterface.php#L25-L26

updated as such below

it's a tiny improvement, but worth clarifying IMHO to rely on the root `\Locale` (either the real one, or the `Intl\Locale\Locale` stub.

`Intl\Locale` extends from `\Locale` and doesnt own the default, only the default fallback.

This internal, first citizent, `Intl\Locale` is super confusing, and is really a `LocaleFallback` util

**Comment by fabpot** [comment]
@jakzal Are you ok to merge this one?

**Comment by ro0NL** [comment]
the sole purpose of the PR is to reduce merge conflicts, if it doesnt matter much we can close (but in master / #31365 we take a different apprach, as im really skeptical about maintaining the lists of consts :D)

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Pull request merged**